### ISSUE TITLE
How should this project be used

### ISSUE THREAD

**Issue by wutong1027** [ashtanmistal/Forest-Friends#15]
Very interesting project! But how should I use it? I haven't seen any way to use it, please help me.
: )
I have a bunch of PCD point cloud files and I want to extract a single tree.

**Comment by ashtanmistal** [comment]
Hi, thanks for reaching out! A few things about the project:
- It was written to supplement another [project of mine](https://github.com/ashtanmistal/minecraftUBC), so a lot of the code is tailored specifically to this. It was completed as part of a sprint leading up to CVPR last year (and looking at it now, has not been cleaned up since then - oops). You'll need to tailor it to your own needs accordingly if you want to use it.
- The [preprocessing_python folder](https://github.com/ashtanmistal/Forest-Friends/tree/main/preprocessing_python#clustering) contains implementation details and is where I would recommend visiting.
- All of the C++ code under the ForestFriends directory was incomplete and should be ignored - I ran out of time to do a better-than-Python implementation :(

You'll want to start with a point cloud dataset that has already been classified (e.g. you should know what points have a "tree" label).

This project 1) clusters points into individual trees, and 2) determines what type of tree each cluster is. 
The first step is in [preprocessing_python/clustering.py](https://github.com/ashtanmistal/Forest-Friends/blob/main/preprocessing_python/clustering.py) - modify `main()` to fit your needs.
For the second step, the trained model is unfortunately not available - you'd need to provide your own training data, but my modified version of PointNet++ is available in the classification folder. So for a few of the clusters determined in the previous step you would need some ground truth data. I used a [dataset from UBC](https://github.com/UBCGeodata/ubc-geospatial-opendata) for this task.

**Comment by wutong1027** [comment]
Okay, got it
Thanks for your reply : ))
